internal/certgen/certgen.go

```diff
@@ -60,11 +69,15 @@ type Config struct {
 	OutputPEM bool
 }
 
-// ContourCerts holds all three keypairs required, the CA, the Contour, and Envoy keypairs.
-type ContourCerts struct {
-	CA      *keyPair
-	Contour *keyPair
-	Envoy   *keyPair
+type certData struct {
+	Filename string
```

Comment:
I think that Filename can probably be inferred from the context in which the certificate is used.

Reply:
This is because, at least in the k8s secret context, certData.Filename is strongly correlated with the name (key) of the secret inside the secret object, and also with the names in deployment yamls, while you _could_ change the filename certData is written too, in reality, people probably won't and I sure don't want to have to document all the places where they need to chase it.